Assets/Mirror/Editor/Weaver/Weaver.cs

```diff
@@ -385,25 +387,33 @@ static bool CheckMessageBase(TypeDefinition td)
 
             bool didWork = false;
 
-            // are ANY parent classes MessageBase
-            TypeReference parent = td.BaseType;
-            while (parent != null)
+            if (td.ImplementsInterface(IMessageBaseType))
             {
-                if (parent.FullName == MessageBaseType.FullName)
-                {
-                    MessageClassProcessor.Process(td);
-                    didWork = true;
-                    break;
-                }
-                try
-                {
-                    parent = parent.Resolve().BaseType;
-                }
-                catch (AssemblyResolutionException)
+                MessageClassProcessor.Process(td);
+                didWork = true;
+            }
+            else
```

Comment:
This else is basically a no-op.  There is no case where it would do anything at all.   

If it is a class derives from MessageBase then the if would enter and call the MessageClassProcessor because MessageBase implements the interface.

If it is a class that implements the inteface,  the if would also enter.

If it is a class/struct that does not implement the interface,  then the else would enter,  but the if in line 401 would never enter,  so the else would not do anything at all.

Please just get rid of the entire else